spec/unit/typesafe/config/tokenizer_spec.rb

```diff
@@ -0,0 +1,696 @@
+require 'spec_helper'
+require 'hocon'
+require 'test_utils'
+require 'pp'
+
+
+describe Hocon::Impl::Tokenizer do
+  Tokens = Hocon::Impl::Tokens
+
+  shared_examples_for "token_matching" do
```

Comment:
I'm really glad you're using this example everywhere, it's going to make porting the lossless token tests much much easier!